packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -433,6 +467,165 @@ export class GraphQLApi extends Construct {
+    return this.grantType(grantee, 'Subscription', fields);
+  }
+
+  /**
+   * Adds an IAM policy statement for Type access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param type The type to grant access
+   * @param fields The fields to grant access to that are of @param type (leave blank for all)
+   */
+  public grantType(grantee: IGrantable, type: string, fields?: [string]): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions: ['appsync:graphql'],
+      resourceArns: fields ?
+        [...fields.map(field => Lazy.stringValue({
+          produce: () => {
+            return this.generateResourceArn({ type: type, field: field });
+          },
+        }))] :
+        [
+          this.generateResourceArn({ type: type }),
+        ],
+      scope: this,
+    });
+  }
+
+  /**
+   * Generate a Resource ARN to use for grant functions.
+   *
+   * If custom is specified, then resourceName is custom.
+   * If type is empty, generate full access ('*').
+   * If field is empty, generate full access to specified type.
+   *
+   * @param resource - an singular IAM resource to provision
+   */
+  private generateResourceArn(resource: IamResources): string {
+    const name = resource.custom ? resource.custom : (
+      resource.type ? resource.field ? `types/${resource.type}/fields/${resource.field}` : `types/${resource.type}/*` : '*'
+    );
+    return Stack.of(this).formatArn({
+      service: 'appsync',
+      resource: `apis/${this.apiId}`,
+      sep: '/',
+      resourceName: `${name}`,
+    });
+  }
+
+  private formatOpenIdConnectConfig(
+    config: OpenIdConnectConfig,
+  ): CfnGraphQLApi.OpenIDConnectConfigProperty {
+    return {
+      authTtl: config.tokenExpiryFromAuth,
+      clientId: config.clientId,
+      iatTtl: config.tokenExpiryFromIssue,
+      issuer: config.oidcProvider,
+    };
+  }
+
+  private formatUserPoolConfig(
+    config: UserPoolConfig,
+  ): CfnGraphQLApi.UserPoolConfigProperty {
+    return {
+      userPoolId: config.userPool.userPoolId,
+      awsRegion: config.userPool.stack.region,
+      appIdClientRegex: config.appIdClientRegex,
+      defaultAction: config.defaultAction || 'ALLOW',
+    };
+  }
+
+  private createAPIKey(config: ApiKeyConfig) {
+    let expires: number | undefined;
+    if (config.expires) {
+      expires = new Date(config.expires).valueOf();
+      const days = (d: number) =>
+        Date.now() + Duration.days(d).toMilliseconds();
+      if (expires < days(1) || expires > days(365)) {
+        throw Error('API key expiration must be between 1 and 365 days.');
+      }
+      expires = Math.round(expires / 1000);
+    }
+    const key = new CfnApiKey(this, `${config.name || 'DefaultAPIKey'}ApiKey`, {
+      expires,
+      description: config.description || 'Default API Key created by CDK',
+      apiId: this.apiId,
+    });
+    return key.attrApiKey;
+  }
```

Comment:
Any reason you moved these in this PR? They seem unchanged.

Reply:
Can you un-move it then, please? We generally try to avoid unrelated changes to files in PRs (makes it quite a bit harder to review them otherwise).